server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMailboxesUpdateProcessor.java

```diff
@@ -210,6 +209,9 @@ private boolean mustChangeParent(Optional<MailboxId> currentParentId, MailboxId
     private void updateMailbox(Mailbox mailbox, MailboxUpdateRequest updateRequest, MailboxSession mailboxSession) throws MailboxException {
         MailboxPath originMailboxPath = mailboxManager.getMailbox(mailbox.getId(), mailboxSession).getMailboxPath();
         MailboxPath destinationMailboxPath = computeNewMailboxPath(mailbox, originMailboxPath, updateRequest, mailboxSession);
+        if (updateRequest.getSharedWith().isPresent()) {
```

Comment:
please use ifPresent instead

Reply:
It makes MailboxException complicated and is not consistant with the rest of the method. Maybe enough refactoring for one ticket?